docs/runbooks.md

```diff
@@ -0,0 +1,147 @@
+- When in `Ignore` mode the Kubernetes API server ignores a webhook failure if the webhook can not be reached
+- This would result in the zone aware pod disruption budget not being enforced or the ZPDB configuration validator not being enforced if the `rollout-operator` is not running
+
+How to **investigate**:
+
+- Review the configuration of the `pod-eviction` and `zpdb-configuration` webhooks:
+  - `kubectl -n <namespace> get ValidatingWebhookConfigurations zpdb-validation-<namespace> -o yaml`
+  - `kubectl -n <namespace> get ValidatingWebhookConfigurations pod-eviction-<namespace> -o yaml`
+- Update the configuration to use a `Fail` policy. See jsonnet configuration options `ignore_rollout_operator_zpdb_eviction_webhook_failures` and `ignore_rollout_operator_zpdb_validation_webhook_failures` in [rollout-operator.libsonnet](https://github.com/grafana/rollout-operator/blob/main/operations/rollout-operator/rollout-operator.libsonnet).
+
+### BadZoneAwarePodDisruptionBudgetConfiguration
+
+This alert fires when the zone aware pod disruption budget configuration validating webhook observes an invalid ZPDB object.
+
+This indicates that a malformed configuration has been applied, or an invalid configuration already exists when the `rollout-operator` starts.
+
+How it **works**:
+
+- If the `zpdb-validation` `ValidatingWebhookConfiguration` has a failure policy set to `Fail`, then the invalid configuration will be rejected by this webhook. This alert will NOT be triggered in this circumstance
+- If the failure policy is set to `Ignore`, then an invalid configuration may have been applied. This can occur if the `rollout-operator` pod is not running and a `ZPDB` configuration is applied. With a failure policy set to `Ignore` the configuration will be accepted if the webhook is unavailable
+- If the `zpdb-validation` `ValidatingWebhookConfiguration` is not applied then an invalid configuration may be applied. This alert will trigger in this scenario.
+
+How to **investigate**:
+
+- Review the `zpdb-validation` `ValidatingWebhookConfiguration` and ensure its failure policy is set to `Fail`
+- `kubectl -n <namespace> get ValidatingWebhookConfigurations zpdb-validation-<namespace> -o yaml`
+- Review the `ZPDB` configurations and verify they are valid
+- `kubectl -n <namespace> get zpdb <name> -o yaml`
+
+### HighNumberInflightZpdbRequests
+
+This alert fires when there has been a sustained number of in-flight pod eviction requests for a given period of time.
+
+This indicates that there is likely an issue causing a delay in the pod eviction consideration process.
+
+How it **works**:
+
+- The `pod-eviction` `ValidatingWebhookConfiguration` routes voluntary pod eviction requests into the ZPDB eviction controller
+- The rollout controller uses the ZPDB eviction controller to test if a pod can be updated (as part of rolling updates)
```

Comment:
Let's make it clearer that this is an independent thing (ie. these first two points aren't a sequence of steps)

```suggestion
- The rollout controller also uses the ZPDB eviction controller to test if a pod can be updated (as part of rolling updates)
```